gcp-devel/main.go

```diff
@@ -0,0 +1,333 @@
+package main
+
+import (
+	"bytes"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"log"
+	"os/user"
+	"strings"
+
+	"github.com/pulumi/pulumi-gcp/sdk/v5/go/gcp/compute"
+	"github.com/pulumi/pulumi-gcp/sdk/v5/go/gcp/container"
+	"github.com/pulumi/pulumi-gcp/sdk/v5/go/gcp/secretmanager"
+	"github.com/pulumi/pulumi-gcp/sdk/v5/go/gcp/serviceaccount"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi/config"
+	"golang.org/x/crypto/ssh"
+)
+
+// DefaultNamePrefix is the default prefix for resource names.
+var DefaultNamePrefix string
+
+type ConfigClustersKubernetes struct {
+	Channel string
+	Version string
+}
+
+type ConfigClustersNodeConfig struct {
+	MachineType string
+	Preemptible bool
+	OauthScopes []string
+}
+
+type ConfigClusters struct {
+	Kubernetes    ConfigClustersKubernetes
+	NetworkPolicy bool
+	NodeLocations []string
+	NodeConfig    ConfigClustersNodeConfig
+	Names         []string
+}
+
+type Config struct {
+	Clusters ConfigClusters
+	Location string
+}
+
+// GenerateSSHKeys ...
```

Comment:
Nit: Add a TODO to collapse this in to the keygen code from aws-devel

Reply:
Addressed